too many warning happens on Jetson TK1

##### System information (version)

- OpenCV => master ( d15d9f5 )
- Operating System / Platform => Ubuntu 14.04 Jetson TK1 (ARM 32bit) (L4T 21.3)
- Compiler => GCC 4.8.4 + CUDA 6.5

##### Detailed description

 - I get dozen of warnings when building OpenCV master on Jetson TK1
```
/usr/local/cuda/include/math_functions.h:255:5: warning: "_MSC_VER" is not defined [-Wundef]
 #if _MSC_VER < 1800
     ^
/usr/local/cuda/include/math_functions.h:275:5: warning: "_MSC_VER" is not defined [-Wundef]
 #if _MSC_VER < 1800
     ^
```
 - This direct cause comes from inside of CUDA 6.5, so I think it's better just to disable this warning
 - I'll create a PR to suppress this warnings

##### Steps to reproduce

1. checkout master
2. ```cmake``` and ```make``` as usual

--- a/modules/superres/CMakeLists.txt
+++ b/modules/superres/CMakeLists.txt
@@ -3,6 +3,9 @@ if(IOS OR WINRT)
 endif()
 
 set(the_description "Super Resolution")
+if(HAVE_CUDA)
+  ocv_warnings_disable(CMAKE_CXX_FLAGS -Wundef -Wshadow)
+endif()
 ocv_define_module(superres opencv_imgproc opencv_video
                   OPTIONAL opencv_videoio opencv_cudaarithm opencv_cudafilters opencv_cudawarping opencv_cudaimgproc opencv_cudaoptflow opencv_cudacodec
                   WRAP python)

